src/python/psic/cnn_model/sample_model_with_training.py

```diff
@@ -0,0 +1,158 @@
+
+# One-hot encode the labels
+y_train = tf.keras.utils.to_categorical(y_train, 10)
+y_valid = tf.keras.utils.to_categorical(y_valid, 10)
+y_test = tf.keras.utils.to_categorical(y_test, 10)
+
+# Print training set shape
+print("x_train shape:", x_train.shape, "y_train shape:", y_train.shape)
+
+# Print the number of training, validation, and test datasets
+print(x_train.shape[0], 'train set')
+print(x_valid.shape[0], 'validation set')
+print(x_test.shape[0], 'test set')
+
+
+model = tf.keras.Sequential()
+
+# CONV 1 64 3x3 filters at stride 1, pad 1
+model.add(tf.keras.layers.Conv2D(filters=64, kernel_size=3, strides=1, padding='same', activation='relu', input_shape=(28,28,1)))
+
+# BN 1
+model.add(tf.keras.layers.BatchNormalization())
+
+# MAX POOL 1
+model.add(tf.keras.layers.MaxPooling2D(pool_size=2))
+model.add(tf.keras.layers.Dropout(0.5))
+
+# CONV 2
+model.add(tf.keras.layers.Conv2D(filters=64, kernel_size=3, strides=1, padding='same', activation='relu'))
+
+# BN 2
+model.add(tf.keras.layers.BatchNormalization())
+
+# MAX POOL 2
+model.add(tf.keras.layers.MaxPooling2D(pool_size=2))
+model.add(tf.keras.layers.Dropout(0.5))
+
+# CONV 3
+model.add(tf.keras.layers.Conv2D(filters=64, kernel_size=3, strides=1, padding='same', activation='relu'))
+
+# BN 3
+model.add(tf.keras.layers.BatchNormalization())
+
+# MAX POOL 3
+model.add(tf.keras.layers.MaxPooling2D(pool_size=2))
+model.add(tf.keras.layers.Dropout(0.5))
+
+model.add(tf.keras.layers.Flatten())
+model.add(tf.keras.layers.Dense(256, activation='relu'))
+model.add(tf.keras.layers.Dropout(0.5))
+model.add(tf.keras.layers.Dense(10, activation='softmax'))
+
+# Take a look at the model summary
+model.summary()
+
+
+# Compile the model
+model.compile(loss='categorical_crossentropy',
+             optimizer='adam',
+             metrics=['accuracy'])
+
+
+# Train the model
+model.fit(x_train,
+         y_train,
+         batch_size=64,
+         # Change to 20 epochs
+         epochs=20,
+         validation_data=(x_valid, y_valid),
+         )
+
+
+#Test Accuracy
+# Evaluate the model on test set
+score = model.evaluate(x_test, y_test, verbose=0)
+
+# Print test accuracy
+print('\n', 'Test accuracy:', score[1])
+
+
+# Visualize prediction
+y_hat = model.predict(x_test)
+
+# Plot a random sample of 10 test images, their predicted labels and ground truth
+figure = plt.figure(figsize=(20, 8))
+for i, index in enumerate(np.random.choice(x_test.shape[0], size=15, replace=False)):
+    ax = figure.add_subplot(3, 5, i + 1, xticks=[], yticks=[])
+    # Display each image
+    ax.imshow(np.squeeze(x_test[index]))
+    predict_index = np.argmax(y_hat[index])
+    true_index = np.argmax(y_test[index])
+    # Set the title for each image
+    ax.set_title("{} ({})".format(fashion_mnist_labels[predict_index],
+                                  fashion_mnist_labels[true_index]),
+                                  color=("green" if predict_index == true_index else "red"))
```

Comment:
@JWeb56 I know that this is just a stub using the mnist intro, but be aware that there are many formatting errors listed by @pep8speaks in the comments of this document that need to be corrected before submitting to master!

Reply:
No, they don't need to be fixed to be committed to master. You WANT them to be fixed, but there are no errors whatsoever with this code.